### PR TITLE
Task3.2 levytskyi

### PR DESCRIPTION
**Pull request by Nikolya9192**
Task 3.2 Done.

### CODE REVIEW

File: src/app/presentation/components/add-workspace/add-workspace.component.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.container {
+    background:linear-gradient(45deg,#161A27, #48537a)
+}
+.addws_title {
+    font-size: 60px !important;
+    font-weight: 800 !important;
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.css]: Don't use !important;

File: src/app/presentation/components/add-workspace/add-workspace.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                  <mat-label>Workspace Name</mat-label>
+                  <input matInput placeholder="Input a name of workspace" maxlength="70" value="" formControlName="Name" required>
+                </mat-form-field>
+
+                <mat-form-field class="addws-description" appearance="fill">
+                  <mat-label>Description</mat-label>
+                  <textarea matInput placeholder="Input a description of workspace" formControlName="Description" required></textarea>
+                </mat-form-field>
+
+                <button style="margin-top: 0.4cm;" mat-flat-button color="primary" class="w-100">Create</button>
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.html]: Let's add a new class and use class for style

File: src/app/presentation/components/add-workspace/add-workspace.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  <mat-grid-list cols="1" rowHeight="100vh">
+      <mat-grid-tile>
+          <div class="addws_form">
+              <form class="addws-form" [formGroup]="addwsform" novalidate (ngSubmit)="submit()">                
+                <p class="addws_form_title">Create workspace</p>
+               
+                <mat-form-field class="addws-name" appearance="fill">
+                  <mat-label>Workspace Name</mat-label>
+                  <input matInput placeholder="Input a name of workspace" maxlength="70" value="" formControlName="Name" required>
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.html]: Same as down

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          if(err.error.errors && typeof err.error.errors === 'object'){
+            const errors = err.error.errors;
+
+            for(let key in errors){
+              for(let indexError in errors[key]){
+                errorMessage += errors[key][indexError] + '\n';
+              }
+            }
+            
+          this.showAlert(errorMessage);
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: Is alert the best solution ?

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+            return;
+          } 
+
+          if(err.error && typeof err.error === 'object'){
+            errorMessage += err.error.error;
+
+            this.showAlert(errorMessage);
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: Same as previous

File: src/app/presentation/components/workspace-list/workspace-list.component.html [hunk 6]
```diff
@@ -2,7 +2,7 @@
   <div class="container">
     <div style="width: 400px;">
       <p>Create New workspace</p>
-      <a mat-raised-button class="link-button" color="primary" href="./addworkspace">Create New</a>
+      <a mat-raised-button class="link-button" color="primary" [routerLink]="['../create']">Create New</a>
     </div>
   </div>
 </section>
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-list/workspace-list.component.html]: Empty line in end of file

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  submit()
+  {  
+    if(this.addwsform.valid){
+      this.createWorkspace = Object.assign({}, this.addwsform.value);
+      this.service.CreateWorkspace(this.createWorkspace).subscribe(
+        () => {
+          Swal.fire({
+            position: 'top-end',
+            icon: 'success',
+            title: '',
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: Why title is empty ?

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            text: "Workspace created successful",
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: Could we move text messages in outer class and reuse them at each place ?

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+export class AddWorkspaceComponent implements OnInit {
+  addwsform : FormGroup;
+  createWorkspace : CreateWorkspace = new CreateWorkspace();
+  
+
+   
+  constructor(private fb:FormBuilder, private service: WorkspaceService, private router: Router) { 
+    this.addwsform=fb.group({        
+      "Name": ["",[Validators.required]],
+      "Description" : ["",[Validators.required]]
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: Why first parameters are empty ?

File: src/app/presentation/components/add-workspace/add-workspace.component.css [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    color: white;
+    text-align: center;
+    text-shadow: 0px 3px 0px #6e67a9, 0px 14px 10px rgb(0 0 0 / 15%), 0px 24px 2px rgb(0 0 0 / 10%), 0px 34px 30px rgb(0 0 0 / 10%);
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.css]: Line length shouldn't be more then 100 characters

File: src/app/presentation/components/add-workspace/add-workspace.component.html [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+                </mat-form-field>
+
+                <button class="w-100" mat-flat-button color="primary" >Create</button>
+              </form>
+
+        </div> 
+
+      </div>    
+  </div>
+</div>
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.html]: Empty line at end of file

File: src/app/presentation/components/add-workspace/add-workspace.component.html (hunk 3 above)
Comment (YevheniiVovk) [reply src/app/presentation/components/add-workspace/add-workspace.component.html]: You don't add class at 11 line

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+export class AddWorkspaceComponent implements OnInit {
+  addwsform : FormGroup;
+  createWorkspace : CreateWorkspace = new CreateWorkspace();
+  
+
+   
+  constructor(private fb:FormBuilder, private service: WorkspaceService, private router: Router) { 
+    this.addwsform=fb.group({        
+      "Name": [null,[Validators.required]],
+      "Description" : [null,[Validators.required]]
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts]: What is the first parameter and why it should be equal to null ?

File: src/app/presentation/components/add-workspace/add-workspace.component.ts (hunk 9 above)
Comment (Nikolya9192) [reply src/app/presentation/components/add-workspace/add-workspace.component.ts]: Becouse, if I write for example ["test",[Validators.required]],then after start page we have this "test" just in the field, and when we want to write our name we have to delete this "test" and after that we write our name. But if we have ["",[Validators.required]] we at once write our name. I think it wiil more simple for user

File: src/app/presentation/components/workspace-list/workspace-list.component.html (hunk 6 above)
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-list/workspace-list.component.html]: Empty line at end of file

File: src/app/presentation/components/add-workspace/add-workspace.component.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { WorkspaceService } from 'src/app/core/services/workspace.service';
+import { Router } from '@angular/router';
+import { CreateWorkspace } from 'src/app/core/models/workspace';
+
+
+
```
Comment (YevheniiVovk) [inline src/app/presentation/components/add-workspace/add-workspace.component.ts:8]: Please leave just one empty line here same at 18-21 lines

Comment (Nikolya9192) [reply src/app/presentation/components/add-workspace/add-workspace.component.ts:8]: ok

File: src/app/presentation/components/workspace-list/workspace-list.component.html (hunk 6 above)
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-list/workspace-list.component.html]: Add empty line to end of file, please

**Review by Sergey988 — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**